app/Decsys/Controllers/WordlistsController.cs

```diff
@@ -0,0 +1,48 @@
+using AutoMapper;
+using Decsys.Auth;
+using Decsys.Data.Entities;
+using Decsys.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace Decsys.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class WordlistsController : ControllerBase
+    {
+        private readonly WordlistService _service;
+        private readonly IMapper _mapper;
+
+        public WordlistsController(WordlistService service, IMapper mapper)
+        {
+            _service = service;
+            _mapper = mapper;
+        }
+
+
+        [HttpPost]
+        [Authorize(Policy = nameof(AuthPolicies.IsSurveyAdmin))]
+        [SwaggerOperation("Get or create a wordlist for the current user")]
```

Comment:
I don't think a POST method should 'get' a wordlist. That's what GET requests are for. Instead of `Ok` use 'CreatedAtAction` (https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.createdataction?view=aspnetcore-6.0)

Reply:
Yeah, this isn't the most restful interface, but it's temporarily this way for a reason.

Ordinarily, yes, you'd have a GET request to 'get' a wordlist by its id.

And you'd have a POST request perhaps to create a new wordlist, and it would be highly appropriate to use `CreatedAtAction()` to return `HTTP 201 Created` with a link to the resource where you can GET it.

Sadly, none of that really applies here.

We are temporarily rolling the two bits of functionality into one endpoint, because the behaviour we want is:
  - Create if not exists
  - Always return it, whether it was created now or previously.

Because of this, it is a "get" action really, but not of a restful resource (we don't include the id because we don't know if there even is one).

Because it has potential non idempotent side effects, POST is the safest verb, despite the non restful behaviour.

Because we don't always create, we currently always return `200 OK`. I'm open to conditionally returning `201 Created` when that's true, but there's no material gain from a consumer's perspective; they don't care about differentiating between it existing prior to this point or not; they just care about getting it.

Basically you're totally correct, but I'm not proposing we do anything about it :D

Later iterations of this functionality will support multiple wordlists with unique id's and then everything will behave restfully.

Reply:
Thanks for the explanation :)